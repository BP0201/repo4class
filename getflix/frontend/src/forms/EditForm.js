import React, { useState } from "react";
import { useNavigate, useParams } from "react-router-dom";
import Alert from "../other/Alert";

function EditForm({ updateUser }) {
  const navigate = useNavigate();
  const { username } = useParams();
  const [formData, setFormData] = useState({
    username,
    password: "",
  });
  const [formErrors, setFormErrors] = useState([]);

  // console.debug(
  //     "LoginForm",
  //     "login=", typeof login,
  //     "formData=", formData,
  //     "formErrors", formErrors,
  // );

  // called when form is submitted
  async function handleSubmit(evt) {
    // prevent page refresh
    evt.preventDefault();
    let result = await updateUser(formData);
    if (result.success) {
      // redirect if successful
      navigate("/");
    } else {
      setFormErrors(formErrors => [...formErrors, "Something went wrong"]);
    }
  }

  // updates form data state when value of inputs change
  function handleChange(evt) {
    const { name, value } = evt.target;
    setFormData(l => ({ ...l, [name]: value }));
  }

  return (
      <div className="EditForm">
        <div className="container col-md-6 offset-md-3 col-lg-4 offset-lg-4">
          <h2 className="my-4">Edit Profile</h2>

          <div className="card shadow">
            <div className="card-body">
              <form onSubmit={handleSubmit}>

                <div className="form-group">
                  <label htmlFor="password">New Password</label>
                  <input
                      id="password"
                      type="password"
                      name="password"
                      className="form-control"
                      value={formData.password}
                      onChange={handleChange}
                      autoComplete="current-password"
                      required
                  />
                </div>

                {formErrors.length
                    ? <Alert type="danger" messages={formErrors} />
                    : null}

                <button
                    data-testid="edit-form-btn"
                    className="btn btn-success float-right"
                    onSubmit={handleSubmit}
                >
                  Update
                </button>
              </form>
            </div>
          </div>
        </div>
      </div>
  );
}

export default EditForm;
