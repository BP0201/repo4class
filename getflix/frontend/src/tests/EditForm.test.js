import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import EditForm from '../forms/EditForm';

describe("EditForm", () => {
  it("should match the snapshot", () => {
    const { asFragment } = render(
      <MemoryRouter>
        <EditForm />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
    expect(screen.getByLabelText("New Password")).toBeInTheDocument();
  });
});