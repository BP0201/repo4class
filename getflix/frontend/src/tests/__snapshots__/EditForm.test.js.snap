// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditForm should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="EditForm"
  >
    <div
      class="container col-md-6 offset-md-3 col-lg-4 offset-lg-4"
    >
      <h2
        class="my-4"
      >
        Edit Profile
      </h2>
      <div
        class="card shadow"
      >
        <div
          class="card-body"
        >
          <form>
            <div
              class="form-group"
            >
              <label
                for="password"
              >
                New Password
              </label>
              <input
                autocomplete="current-password"
                class="form-control"
                id="password"
                name="password"
                required=""
                type="password"
                value=""
              />
            </div>
            <button
              class="btn btn-success float-right"
              data-testid="edit-form-btn"
            >
              Update
            </button>
          </form>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
